activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/specification/SpecificationSupport.java

```diff
@@ -0,0 +1,252 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.cloud.services.query.rest.specification;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Path;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import jakarta.persistence.metamodel.SingularAttribute;
+import java.util.Collection;
+import java.util.Set;
+import org.activiti.cloud.services.query.model.ProcessVariableEntity;
+import org.activiti.cloud.services.query.model.ProcessVariableEntity_;
+import org.activiti.cloud.services.query.model.dialect.JsonValueFunctions;
+import org.activiti.cloud.services.query.rest.exception.IllegalFilterException;
+import org.activiti.cloud.services.query.rest.filter.VariableFilter;
+import org.springframework.data.jpa.domain.Specification;
+
+public abstract class SpecificationSupport<T> implements Specification<T> {
+
+    protected void addLikeFilters(
+        Collection<Predicate> predicates,
+        Set<String> valuesToFilter,
+        Root<T> root,
+        CriteriaBuilder criteriaBuilder,
+        SingularAttribute<T, String> attribute
+    ) {
+        predicates.add(
+            valuesToFilter
+                .stream()
+                .map(value ->
+                    criteriaBuilder.like(criteriaBuilder.lower(root.get(attribute)), "%" + value.toLowerCase() + "%")
+                )
+                .reduce(criteriaBuilder::or)
+                .orElse(criteriaBuilder.conjunction())
+        );
+    }
+
+    protected Predicate[] getProcessVariableValueFilters(
+        Root<ProcessVariableEntity> root,
+        Collection<VariableFilter> filters,
+        CriteriaBuilder criteriaBuilder
+    ) {
+        return filters
+            .stream()
+            .map(filter ->
+                criteriaBuilder.and(
+                    criteriaBuilder.equal(
+                        root.get(ProcessVariableEntity_.processDefinitionKey),
+                        filter.processDefinitionKey()
+                    ),
+                    criteriaBuilder.equal(root.get(ProcessVariableEntity_.name), filter.name()),
+                    getVariableValueCondition(root.get(ProcessVariableEntity_.value), filter, criteriaBuilder)
+                )
+            )
+            .toArray(Predicate[]::new);
+    }
+
+    protected Predicate getVariableValueCondition(
+        Path<?> valueColumnPath,
+        VariableFilter filter,
+        CriteriaBuilder criteriaBuilder
+    ) {
+        return criteriaBuilder.isTrue(
+            switch (filter.operator()) {
+                case EQUALS -> switch (filter.type()) {
```

Comment:
instead of having a case of cases, can we replace something like a strategy? Where we have a class with one strategy for each type (booelan, integer, ...), of course an interface, and the stratehy selector here. Thanks!